src/porepy/models/constitutive_laws.py

```diff
@@ -2747,6 +2751,11 @@ def gravity_force(
             rho = self.fluid.density(cast(pp.SubdomainsOrBoundaries, grids))
         elif material == "solid":
             rho = self.solid_density(grids)
+        elif material == "bulk":
+            phi = self.porosity(grids)
+            rho_f = self.fluid.density(cast(pp.SubdomainsOrBoundaries, grids))
```

Comment:
Is the typing of grids meaningful? I suspect the signature type does not make sense.

Reply:
I believe I do not understand. Are you referring to line 2731 or 2756?

Reply:
2731, really.

Reply:
I did not make changes here. I am not sure. Does gravity enter Darcy's law on the mortar grid? Then interface grids are relevant. Or what are you concerned about? Maybe I am missing your point.

Reply:
This will be addressed as a separate issue.